internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -4793,6 +4806,14 @@ func expandKubernetesClusterServiceMeshProfile(input []interface{}, existing *ma
 		}
 
 		profile.Istio.Components.IngressGateways = &istioIngressGatewaysList
+
+		if existing != nil && existing.Istio != nil && existing.Istio.Revisions != nil {
+			profile.Istio.Revisions = existing.Istio.Revisions
```

Comment:
What is this for?

Reply:
Previously, if the service_mesh_profile is set with istio, it will be deployed fine, but any updates for service_mesh_profile properties later will cause error returned by the AKS API because the revisions has not been specified. 
I have added this block so that if user wants to update the service_mesh_profile without revisions, it will by default use whatever revisions it already has as returned by the AKS API.

Reply:
Based on this it sounds like this property might actually want to be `Required` instead of `Optional`

Reply:
@stephybun Updated the field to Required, let me know any other changes needed.
As this will be a breaking change, I have made it available in v4.0 and removed the contents from doc and updated the code.

When version 4.0 is released, I will:
- Add back the contents to the doc (or should I still add it & indicate it is only required in v4.0 and on)
- Update the new tests and existing tests to also include this Required field
- Add revisions back as a property in the data source

Reply:
@stephybun 👍🏻👍🏻

Updated the existing test cases config with the feature flag.

Will add `revisions` to data source together in 4.0 release because it shares the same read function with the resource, so it would not work given that the resource doesn't have this property yet.